KBKeyboardObserver/KBKeyboardObserver.m

```diff
@@ -104,6 +144,72 @@ - (void)keyboardDidHide:(NSNotification*)aNotification
     }
 }
 
+- (void)keyboardWillChangeFrame:(NSNotification*)aNotification
+{
+    if (self.interfaceRotation)
+    {
+        return;
+    }
+    
+    NSDictionary *info = aNotification.userInfo;
+    CGRect kbRectEnd = [info[UIKeyboardFrameEndUserInfoKey] CGRectValue];
+    NSTimeInterval duration = [info[UIKeyboardAnimationDurationUserInfoKey] doubleValue] ;
+    CGRect viewKbRectEnd = [self.referenceView convertRect:kbRectEnd fromView:nil];
+    
+    
+    if(CGRectContainsRect(self.referenceView.frame, viewKbRectEnd) && !self.standardKeyboard) {
+        if ([self.delegate respondsToSelector:@selector(keyboardObserver:observedKeyboardWillShowToRect:duration:)]) {
+            [self.delegate keyboardObserver:self
+             observedKeyboardWillShowToRect:viewKbRectEnd
+                                   duration:duration];
+        }
+    } else if (!self.standardKeyboard) {
+        if ([self.delegate respondsToSelector:@selector(keyboardObserver:observedKeyboardWillHideToRect:duration:)]) {
+            [self.delegate keyboardObserver:self
+             observedKeyboardWillHideToRect:viewKbRectEnd
+                                   duration:duration];
+            }
+    }
+    
+    if ([self.delegate respondsToSelector:@selector(keyboardObserver:observedKeyboardDidChangeFrameToRect:duration:)]) {
+        [self.delegate keyboardObserver:self
+   observedKeyboardDidChangeFrameToRect:viewKbRectEnd
+                               duration:duration];
+    }
+    
+}
+
+- (void)keyboardDidChangeFrame:(NSNotification*)aNotification
+{
+    if (self.interfaceRotation)
+    {
+        return;
+    }
+    
+    NSDictionary *info = aNotification.userInfo;
+    CGRect kbRectEnd = [info[UIKeyboardFrameEndUserInfoKey] CGRectValue];
+    CGRect viewKbRectEnd = [self.referenceView convertRect:kbRectEnd fromView:nil];
+    
+    _keyboardVisible = CGRectContainsRect(self.referenceView.frame, viewKbRectEnd);
```

Comment:
One more thing - added delegate methods should be added to the `ExampleViewController`, so anyone could test if the Pod is working. Looking forward for your answers! 😄